src/API/Google/AdsConversionAction.php

```diff
@@ -133,14 +136,17 @@ public function get_conversion_action( $resource_name ): array {
 		} catch ( Exception $e ) {
 			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
 			$message = $e->getMessage();
+			$code    = $e->getCode();
+
 			if ( $e instanceof ApiException ) {
 				$message = $e->getBasicMessage();
+				$code    = $this->map_grpc_code_to_http_status_code( $e );
 			}
 
 			throw new Exception(
```

Comment:
At the moment we are testing only if the Exception is an instance of ApiExeption, we could test the case when the Exception is not an instance of ApiExeption to cover all the possibilities.

Reply:
Thanks, added in ab5b1f5